### ISSUE TITLE
Check (missing) arguments on function call

### ISSUE THREAD

**Issue by yzhang-gh** [microsoft/pylance-release#305]
#### Minimum working example

Say we have such a Python file

```python
def foo(a, b):
   return a + b

foo(1)
```

There is no warning but if we run it, we get

```
Traceback (most recent call last):
  File "main.py", line 4, in <module>
    foo(1)
TypeError: foo() missing 1 required positional argument: 'b'
```

It will be really helpful if Pylance can do this kind of check (just like other static type programming languages).

**Comment by erictraut** [comment]
Pylance will report this (and many other) classes of problems if you enable the type checker. Simply set the "python.analysis.typeCheckingMode" setting to "basic". The type checker is off by default because many Pylance users are interested primarily in completion suggestions and not other errors.

**Comment by yzhang-gh** [comment]
Ah, was wondering it was impossible that a type checker didn't have this feature. Thanks a lot!

**Comment by erictraut** [comment]
Technically, the type checker is always enabled in Pylance, but many of the diagnostic rules are disabled if "typeCheckingMode" is "off". If you switch this setting to "basic", you will enable diagnostic rules that detect many common error conditions like in your example above.

**Comment by yzhang-gh** [comment]
Thanks. I will play around with the configurations 😉.

**Comment by yzhang-gh** [comment]
Well, I started to know why many users would rather turn it off (until I opened a real Python project).

For example

```python
import matplotlib.pyplot as plt

plt.gca().set_xticks([])
          ^^^^^^^^^^
plt.gca().tick_params()
          ^^^^^^^^^^^
```

Pylance does know that `gca` returns an `Axes` object, but it complains about 'Cannot access member "set_xticks" for type "Axes"', which is not true ([link](https://matplotlib.org/3.3.1/api/_as_gen/matplotlib.axes.Axes.set_xticks.html)).

I'm not sure whether it is because Matplotlib is not properly typed or it is a bug of Pylance. But I can still imagine there will be many false alarms due to the various Python packages.

And unfortunately, I cannot disable this kind of error without affecting the case mentioned in the beginning. They are both under the `reportGeneralTypeIssues` rule.

Probably we shouldn't have asked for that much for Python 😥.
